### ISSUE TITLE
Impossible to upload img after migrate server

### ISSUE THREAD

**Issue by JulienNkeys** [PrestaShop/PrestaShop#36526]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Good morning,
We have been looking for a week without success.

We migrated several Prestashops to a server, no problem for all Prestashops except 1.
It is impossible to load an image from the product page, we have a directory error.

We checked the write rights, tests were carried out via outsourcing on the server, we can write to the tmp directory without problem.

The image loads, the thumbnail appears and then this error message in red.
Then when refreshing the thumbnail is white, and if we want to access the directory via the url, there is nothing in it, the directory is not created.

Unable to find the problem.

Prestashop 1.7.8.5

<img width="1192" alt="bug-report" src="https://github.com/PrestaShop/PrestaShop/assets/113473821/070f7ccf-7951-4b73-89bb-0c7ff93bcaa2">

### Steps to reproduce

Adding an image from the manager to a product sheet.

### PrestaShop version(s) where the bug happened

1.7.8.5

### PHP version(s) where the bug happened

7.4

### Your company or customer's name goes here (if applicable).

Newkeys

**Comment by florine2623** [comment]
Hello @JulienNkeys ,

This is a public space, please change your admin for security reason :)

Could you enable debug mode please ?

Thanks!

**Comment by JulienNkeys** [comment]
I have edit capture, thanks :)
When i enable Debug mode i have a problem with another addon, but if i disable it, same problem.

<img width="1367" alt="Capture d’écran 2024-07-09 à 16 59 08" src="https://github.com/PrestaShop/PrestaShop/assets/113473821/61d1b467-9812-46fc-9302-2916b3d96c9f">

**Comment by florine2623** [comment]
Can you disable the module tax135gpb and try again ?
The problems comes from there. You can contact the developer of the module about that issue.

**Comment by JulienNkeys** [comment]
Same problem, we have disable the addon, debug work now.
The error:

CRITICAL17:42:30 | php | Uncaught Exception: The file "/tmp/phpWtEluS" does not existShow context  
CRITICAL17:42:30 | request | Uncaught PHP Exception Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException: "The file "/tmp/phpWtEluS" does not exist" at /var/www/vhosts/*****.com/httpdocs/vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/File/File.php line 37Hide context    Show trace[▼   "exception" => FileNotFoundException {#3720 ▼     #message: "The file "/tmp/phpWtEluS" does not exist"     #code: 0     #file: "/var/www/vhosts/****.com/httpdocs/vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/File/File.php"     #line: 37     

CRITICAL
17:42:30
php	Uncaught Exception: The file "/tmp/phpWtEluS" does not exist

CRITICAL
17:42:30
request	Uncaught PHP Exception Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException: "The file "/tmp/phpWtEluS" does not exist" at /var/www/vhosts/****.com/httpdocs/vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/File/File.php line 37


  "exception" => FileNotFoundException {[#3720 ▼](https://***.com/admin****/index.php/_profiler/e6fe11?panel=logger&_token=D5nGkiJYvOxzzFaDUblQ3-_ewtP46t_JdG2nZmtxu-E#sf-dump-630286161-ref23720)
    #message: "The file "/tmp/phpWtEluS" does not exist"
    #code: 0
    #file: "[/var/www/vhosts/****.com/httpdocs/vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/File/File.php](https://****.com/admin*****/index.php/_profiler/open?_token=D5nGkiJYvOxzzFaDUblQ3-_ewtP46t_JdG2nZmtxu-E?file=vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/File/File.php&line=37#line37)"
    #line: 37
    trace: {▶}
  }

**Comment by matks** [comment]
> We checked the write rights, tests were carried out via outsourcing on the server, we can write to the tmp directory without problem.

Are you sure that the system user used by the webserver has the rights?

For example if you use apache you need to verify the rights of the user running apache.

**Comment by JulienNkeys** [comment]
We use Plesk, we have 5 website Prestashop on this server, and no problem with another websites, only with this. server outsourcing confirms that the rights are ok.

**Comment by florine2623** [comment]
Can you try to delete the cache and make sure you're using the right version of PHP.

https://devdocs.prestashop-project.org/1.7/basics/installation/system-requirements/

**Comment by JulienNkeys** [comment]
Yes it's the good version of php and we have already try to delete cache in /var/

**Comment by SamuelV2C** [comment]
Good morning, same problem here since today

**Comment by Samuel871933** [comment]
Good morning, same problem here since today.
BO cache deleted, Manually cache deleted, php/rights on folders have not been updated, already verified it doesn't come from htaccess too.
Presta version 1.7.8.8 php 7.4.33


```
CRITICAL16:52:02 | php | Uncaught Exception: The file "/tmp/php2k3W7g" does not exist
CRITICAL16:52:02 | request | Uncaught PHP Exception Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException: "The file "/tmp/php2k3W7g" does not exist" at /var/www/vhosts/website.fr/httpdocs/vendor/symfony/symfony/src/Symfony/Component/HttpFoundation/File/File.php line 37
```

**Comment by Samuel871933** [comment]
The problem was PrestaShop EventBus module v3.0.13 for me. Upgrading the module to v3.0.14 seems to fix the problem.

**Comment by JulienNkeys** [comment]
> The problem was PrestaShop EventBus module v3.0.13 for me. Upgrading the module to v3.0.14 seems to fix the problem.

OMG, perfect, it's that.
Thank you very much !

**Comment by matks** [comment]
Thanks for sharing the tip @SamuelV2C ;)

**Comment by regineParis** [comment]
Bonjour, nous somme le 25 aout 2024. J'ai eu un problème sur prestashop 1.7.8, impossible de télécharger de nouvelles images produit depuis le BO. Message d'erreur identique a celui cité plus haut par JulienNkeys.
J'ai désactivé le module enventbus, ce qui a réglé le problème. Puis mise à jour du module (v3.0.13 ==> v3.0.16 ) ce qui a résolu le problème. Merci

**Comment by totti240282** [comment]
Same problem with v4.0.1

**Comment by M-Falken** [comment]
Same problem with v4.0.1
I disabled the enventbus module, which fixed the problem.
May we have a fix for this please ?
Thx

PrestaShop: 1.7.8.11
PHP: 7.4.33

**Comment by atarigamer** [comment]
Same issue after upgrading Event Bus to 4.0.1
PS: 1.7.8.11
PHP: 7.4.33

**Comment by M-Falken** [comment]
Event Bus 4.0.2 is available , some feedback ?

**Comment by M-Falken** [comment]
Some feedback with the 4.0.2 ?

**Comment by atarigamer** [comment]
Provided feedback on the other ticket that I raised - https://github.com/PrestaShopCorp/ps_eventbus/issues/403

**Comment by Hlavtox** [comment]
@PrestaShop/qa-functional Guys can you try to reproduce, put proper labels or close it?

Or if its an issue in eventbus, send it to the company team?

**Comment by boo-code** [comment]
Not reproducible on 9.2.x, and the cause was never in core.

The chain in this thread is `ps_eventbus` consuming the PHP upload temporary file before the controller reads it - reported here for 3.0.13 (fixed in 3.0.14) and again for 4.0.1. The module tracker has it as PrestaShopCorp/ps_eventbus#403, fixed in 4.0.2 on 2025-01-15 and confirmed there by the reporter; current release is 4.1.1.

The core half of the trace is gone as well. `vendor/symfony/symfony/.../File.php:37` is the monolithic `symfony/symfony` package, which 9.2.x does not ship (its `composer.lock` has the split components only), and `controllers/admin/AdminProductsController.php` no longer exists - 30 legacy admin controllers remain and that is not one of them. On 9.2.x a product image upload runs through `ImageController::addImageAction`, which wraps the whole handler in `catch (Exception)` and answers with a 400 JSON payload, so it cannot produce the uncaught 500 in the screenshot.

@Hlavtox - nothing left to reproduce; the module fix is 14 months old, so this can be closed.

One thing found while checking that, which is worth its own report rather than this thread: 9.2.x still reports a vanished upload badly. `ImageValidator::assertFileUploadLimits()` calls `filesize()` before anything asserts the file is there, and every caller runs it ahead of `assertIsValidImageType()`, which is where the existing `ImageFileNotFoundException` lives. Driving the real product image form builder and form handler with an `UploadedFile` whose path has been deleted:

```
submitted=true valid=true
handler THREW InvalidArgumentException: "" cannot be interpreted as a number
```

`(string) filesize($missing)` is `""`, and `ProductImageFileValidator` hands that to `DecimalNumber`. Control, same request with the file still on disk: `UploadedImageConstraintException: Image type "text/plain" is not allowed`. `ImageFileNotFoundException` is unmapped in `getErrorMessages()` of all three controllers that can raise it, so even when it is raised correctly the merchant reads `An unexpected error occurred. [InvalidArgumentException code 0]`. That opacity is what sent the people in this thread auditing directory permissions for a week, and it is a separate PS9 defect - it belongs in its own issue, not here.
